src/components/ListItem.jsx

```diff
@@ -1,5 +1,62 @@
 import './ListItem.css';
+import { useEffect, useState } from 'react';
+import { updateItem } from '../api/firebase';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils/dates';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, listPath, itemId, dateLastPurchased }) {
+	const [isChecked, setIsChecked] = useState(false);
+	const [expired, setExpired] = useState();
+
+	const handleChange = async () => {
+		if (!isChecked) {
+			const userConfirmed = window.confirm('Confirm purchase?');
+			if (userConfirmed) {
+				setIsChecked(!isChecked);
+			}
+		}
+	};
+
+	useEffect(() => {
+		if (dateLastPurchased) {
+			const expiryDate = new Date(
+				dateLastPurchased.seconds * 1000 + ONE_DAY_IN_MILLISECONDS,
+			);
+
+			if (new Date() > expiryDate) {
+				setIsChecked(false);
+				setExpired(false);
+			} else {
+				setIsChecked(true);
+				setExpired(true);
+			}
+		}
+	}, [dateLastPurchased]);
+
+	useEffect(() => {
+		const handleUpdateItem = async () => {
+			if (isChecked) {
+				await updateItem(listPath, itemId);
+			}
+		};
+		try {
+			handleUpdateItem();
+		} catch (error) {
+			console.log('error: ', error);
+		}
+	}, [isChecked]);
```

Comment:
I'm seeing a few missing dependencies from this array:

<img width="879" alt="Screenshot 2024-03-01 at 3 04 39 PM" src="https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/78327387/b479ef93-3730-43fc-89ca-252cea181056">